### PR TITLE
[ZNC] Update to v1.7.1 and generic DSM5+6 service support

### PR DESCRIPTION
**Pull request by m4tt075**
### Motivation
ZNC update request and lack of generic DSM5+6 service support
Linked issues: #2977 

### Checklist
- [x] Build rule `all-supported` completed successfully
- [ ] Package upgrade completed successfully
- [x] New installation of package completed successfully

### Caveats
ZNC 1.7.1 requires `c++11` compiler support, which is only available with newer Synology toolchains. On the other hand, older ZNC packages have issues (see [here](https://github.com/znc/znc/issues/947) for a related discussion).

### CODE REVIEW

**Comment by luckman212** [comment]
Just posting here to report that after running this for 2 days, I have no issues to report. I checked the following logs and see no errors or issues:
```
/volume1/@appstore/znc/var/znc_install.log
/volume1/@appstore/znc/var/moddata/adminlog/znc.log
/volume1/@appstore/znc/var/znc.log
```

**Comment by derfelot** [comment]
Did an upgrade over 1.6.0 for rtd1296 (DS118), looking good so far.

**Comment by m4tt075** [comment]
New test builds available here...
https://github.com/m4tt075/spksrc/releases/tag/znc-1.7.1-4

**Comment by ymartin59** [comment]
@m4tt075 Is there any reason except historical for modules sources to be included here? From my point of view, makefile should grab directly source files from original upstream locations, either invoking git or wget, curl...

**Comment by m4tt075** [comment]
@ymartin59 No reason, just how the package was historically designed. We could `wget` the files from github master trees, but then they are moving targets and we don't know what we get without introducing hash checks. If we `wget` the github releases, we have to extract the cpp modules and patch (at least one of) them. Whatever we do, it is going to be more complicated. Is it worthwhile (with 6 downloads of the test package after 20 days)? Or is this required for some reason?

File: spk/znc/Makefile [hunk 1]
```diff
@@ -1,31 +1,41 @@
...
-MAINTAINER = Diaoul
+MAINTAINER = m4tt075
 DESCRIPTION = Advanced IRC bouncer. An IRC bouncer is nothing more than an IRC proxy. ZNC will always be connected in your chat rooms, and will be the gateway between your clients, and your IRC servers. You can, for instance, consult messages while you were offline or hide your identity.
 DESCRIPTION_FRE = Bouncer IRC avancé. Un bouncer IRC n\’est en fait qu\’un proxy IRC. ZNC restera connecté en permanence à vos salons de discussions et fera office de passerelle entre vos clients et vos serveurs IRC. Vous pourrez, par exemple, consultez des messages diffusés en votre absence ou masquer votre identité.
 DESCRIPTION_SPN = IRC bouncer avanzado. Un IRC bouncer no es más que un proxy para IRC. ZNC estará siempre conectado a tus canales, y hará de puerta de enlace entre tus clientes y tus servidores IRC. Puedes, por ejemplo, consultar mensajes mientras estuviste desconectado o esconder tu identidad.
-ADMIN_PORT = 8250
 RELOAD_UI = yes
 DISPLAY_NAME = ZNC
-CHANGELOG = "1. Update to ZNC 1.6.0<br>2. Update modules<br>3. Add playback module"
+CHANGELOG = "1. Update to ZNC 1.7.1<br>2. Update modules<br>3. Add generic DSM5+6 service support"
```
Comment (ymartin59) [inline spk/znc/Makefile]:
```suggestion
CHANGELOG = "1. Update to ZNC 1.7.1<br/>2. Update modules<br/>3. Add generic DSM5+6 service support"
```

**Comment by ymartin59** [comment]
Why not create a "cross/znc-modules" and use standard build chain support for download+checksum+patching with a variable for modules list to build ?
When application works, there is no specific requirement, I just expect to get maintainer job easier. There is no information where these cpp files come from and how they have been patched... Not easy at all when upgrading upstream version. What do you think about it?

**Comment by m4tt075** [comment]
Sounds nice, but I don't know how to realize that without having to add a cross-package for each and every module. How would I recurse through a list within the Makefile, where sources, dist-files, etc. differ? Also, I've not been able to find a source of the `identd.cpp` module. I just kept the one of the original package.

**Comment by ymartin59** [comment]
My proposal is to replace default "compile" target in cross/znc-modules with custom commands to build modules (in for loop for instance) the "standard" way it is documented. May you please give it a trial?

**Comment by m4tt075** [comment]
@ymartin59 I'd really like to and am aware of loop commands, but still have no idea how. Let me explain why I'm stuck:
1. There are two options to build znc-modules, firstly, in-tree during the `znc` build process, secondly, using a specific `znc-buildmod` command, which is shipped with the `znc` package. [Here](https://wiki.znc.in/Compiling_modules) are details. 
2. With a separate cross-package, I believe in-tree building would be the way to go. For `znc` to find the modules, we just have to copy the `[module].cpp` source files into a certain directory of the `znc` source tree, so that they are picked up during the `znc` build process. 
3. Therefore, the `znc-modules` cross-package would have to do the following, but **looping through the different modules** we want to integrate:
a. Download the module source.tar.gz
b. Verify the hashsum
c. Extract it
d. Copy the extracted module.cpp file to the right directory within the `znc` source tree.

I understand that the downloading, hash-checking  and extraction pieces are only done once per cross/package/Makefile and not multiple times. I also don't think that `mk/download.mk` can handle multiple files at once (especially when they need to be downloaded from multiple sources. Also, the cross-package would not have to configure or compile anything. That is also why I don't understand why a compile_traget would help us here. But no matter which target I'd use, the "code" I'd need is 3 a,b,c,d which is exactly what the framework should do for me. I've been looking for a precedent, but could not find one. Do you know one? The only way I see, is to create a cross-package for each of the six individual modules, but not sure that's what we want. Please advise...

**Comment by ymartin59** [comment]
@m4tt075 Thanks for details. I have just checked that "modules" folder is part of the znc github release archive, so there is no need for "cross/znc-modules". As I fear "znc-buildmod" command may not support cross-compilation, I propose to try to build modules in-tree with znc itself.

**Comment by m4tt075** [comment]
@ymartin59 I've spent three more hours thinking about this yesterday and played around with different options. In my eyes, the current solution is the best, no matter whether we use in-tree building or the "znc-buildmod". The problem is the same: We always have to inject the additional `[module].cpp` files into the `znc` build process (sooner or later). We can use `wget` for that. But then we need hash checks or we lose control about which files we are injecting. In effect that means the following: We need to store hashtags in the package, instead of the actual `[module].cpp", `wget` the modules, compare them to the hashtags, inject them if they match or skip them if they don't. It is easier, safer and less error-prone to just store the actual `[module].cpp` files locally in the first place, so that we know what we are injecting. Of course your initial point is absolutely correct. We have to know where those files are coming from. I've found sources for all modules, and even ones where (I believe) we do not need any patches. I'll include documentation for them in the Makefile, so that every maintainer will be able to see them.

**Comment by m4tt075** [comment]
@ymartin59 OK, pushed the changes. Could you review the approach, please? If you are fine with it, I'd like to produce another round of test-builds before we merge. There are only few users for this package, so we should keep the number of test-build versions low...

File: spk/znc/src/service-setup.sh [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    # Edit the configuration according to the wizard
+    sed -i -e "s,@pidfile@,${PID_FILE},g" ${SYNOPKG_PKGDEST}/var/configs/znc.conf
+    sed -i -e "s,@certfile@,${CERT_FILE},g" ${SYNOPKG_PKGDEST}/var/configs/znc.conf
+    sed -i -e "s,@username@,${wizard_username:=admin},g" ${SYNOPKG_PKGDEST}/var/configs/znc.conf
+    sed -i -e "s,@password@,${wizard_password:=admin},g" ${SYNOPKG_PKGDEST}/var/configs/znc.conf
+    sed -i -e "s,@zncuser@,${EFF_USER},g" ${SYNOPKG_PKGDEST}/var/configs/oidentd.conf
+
+    # Generate certificate
+    echo "Generating initial certificate file" >> ${INST_LOG}
+    su ${EFF_USER} -s /bin/sh -c "${ZNC} -d ${SYNOPKG_PKGDEST}/var -p" >> ${INST_LOG}
```
[the remark this answers is not in the document]
Comment (m4tt075) [reply spk/znc/src/service-setup.sh]: Done.

**Comment by m4tt075** [comment]
Moved certificate file creation and tested the change on my XPE 6.1 system. Ready to go from my perspective. Please feel free to squash merge whenever you deem appropriate...

**Pull request merged**